formal_spec/nmt.qnt

```diff
@@ -610,7 +610,7 @@ module nmtTest {
 
     // this module iteratively generates a proof and then non-deterministically corrupts it.
     // run by:
-    // quint run --main=nmtTest --max-samples=1 --max-steps=100 nmt.qnt --out-itf=ITF_files/out.itf.json
+    // quint run --main=nmtTest --max-samples=2 --max-steps=100 nmt.qnt --out-itf=ITF_files/out.itf.json
```

Comment:
[question] why 1 became 2?

Reply:
I don't remember actually, probably I was copying from the terminal when experimenting with that problem of empty traces. Will return to 1 again. (`--max-samples=2` covers more examples, but takes a bit longer)

Reply:
> Thanks for addressing the comments, they look good to me! I have added some inline comments some of which are related to the CIs failure. In addition to those:
> 
> * In order to avoid any confusion between the already existing spec folder the the one added by this PR, can you please change the name of the `formal_spec` directory to `nmt_quint_model` instead?
> * Our CIs are failing because the current branch is out of sync with the nmt master branch. Can you please merge the latest changes from the master branch into this branch? Thank you.
> * Also, would you please let me know whether CIs are visible on your end?

I renamed the folder, rebased the master branch onto this branch, and fixed the issues you pointed to. Yes, CIs are visible on my end. I can see that `gobencher` is still failing, but I don't understand why. Do you?

Other than that, I believe the PR is ready to be merged.